andhow-core/src/test/java/org/yarnandtail/andhow/internal/ConstructionProblemTest.java

```diff
@@ -0,0 +1,165 @@
+package org.yarnandtail.andhow.internal;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+import org.junit.Test;
+import org.yarnandtail.andhow.api.*;
+import org.yarnandtail.andhow.AndHowInit;
+import java.util.ArrayList;
+import java.util.List;
+import org.yarnandtail.andhow.AndHow;
+
+
+
+public class ConstructionProblemTest {
+
+    @Test    
+    public void testNoUniqueNames() {
+        GroupProxy refGroup = mock(GroupProxy.class);
+        
+        Property refProperty = mock(Property.class);
+        GroupProxy badGroup = mock(GroupProxy.class);
+        Property badProperty = mock(Property.class);
+        String conflictName = "conflict name test";
+        ConstructionProblem.NonUniqueNames instance = new ConstructionProblem.
+        NonUniqueNames(refGroup, refProperty, badGroup, badProperty, conflictName);
+        
```

Comment:
For the `NonUniqueName` problem (and others where there is a ref group and ref property), there is a duplication or clash between two properties and the other property is listed as reference to help the user find the error.  Testing that would look like this:
```suggestion
        assertEquals(refGroup , instance.getRefPropertyCoord().getGroup());
        assertEquals(refProperty , instance.getRefPropertyCoord().getProperty());
```